### ISSUE TITLE
Docs: no-misused-promises example doesn't work in playground

### ISSUE THREAD

**Issue by mx-bernhard** [typescript-eslint/typescript-eslint#8053]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The incorrect example for checksVoidReturn at https://typescript-eslint.io/rules/no-misused-promises/#checksvoidreturn includes a snippet that uses `EventEmitter` but the `EventEmitter` is not available as type and such the rule does not detect any violations:

```ts
const eventEmitter = new EventEmitter();
eventEmitter.on('some-event', async () => {
  synchronousCall();
  await doSomething();
  otherSynchronousCall();
});
```

![grafik](https://github.com/typescript-eslint/typescript-eslint/assets/30629795/f69591b5-ce43-440a-857b-89d7c580c262)

I suggest the following replacement of the example:

```ts
class EventEmitter {
  constructor() { }
  on(eventName: string, callback: () => void) { }
}
const synchronousCall = () => { };
const doSomething = async () => { };
const otherSynchronousCall = () => { };

const eventEmitter = new EventEmitter();
eventEmitter.on('some-event', async () => {
  synchronousCall();
  await doSomething();
  otherSynchronousCall();
});
```

### Affected URL(s)

https://typescript-eslint.io/rules/no-misused-promises/#checksvoidreturn

**Comment by JoshuaKGoldberg** [comment]
Yeah that code snippet was added several years ago in #612, before we had a playground (#4108), let alone open-in-playground links 😄. Accepting PRs to make the things work in the playground.  Thanks for noticing+filing!

It would be nice to simplify the examples though. They're pretty convoluted. Not relying on a class pattern used by only a subset of JS/TS devs would be nice - if it's possible to still get the point across...
